bskyogcard/src/routes/post.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react'
+import {AppBskyFeedPost, AtUri} from '@atproto/api'
+import resvg from '@resvg/resvg-js'
+import {Express} from 'express'
+import satori from 'satori'
+
+import {Post} from '../components/Post.js'
+import {AppContext} from '../context.js'
+import {getModeratorData} from '../data/getModeratorData.js'
+import {getPost} from '../data/getPost.js'
+import {getPostData} from '../data/getPostData.js'
+import {httpLogger} from '../logger.js'
+import {loadEmojiAsSvg} from '../util.js'
+import {handler, originVerifyMiddleware} from './util.js'
+
+const WIDTH = 600
+
+export default function (ctx: AppContext, app: Express) {
+  return app.get(
+    '/:actor/post/:rkey',
+    originVerifyMiddleware(ctx),
+    handler(async (req, res) => {
+      let {actor, rkey} = req.params
+      let uri = AtUri.make(actor, 'app.bsky.feed.post', rkey)
+
+      try {
+        const post = await getPost({
+          uri,
+          agent: ctx.appviewAgent,
+        })
+
+        if (!AppBskyFeedPost.isRecord(post.record)) {
+          return res.status(404).end('not found')
+        }
+        const notPublic = post.author.labels.some(
+          l => l.val === `!no-unauthenticated`,
+        )
+        if (notPublic) {
+          return res.status(404).end('not found')
+        }
```

Comment:
If user disables PWI, then we just 404 here. I figured why waste compute on an image of nothing.